inc/priam/cluster.hpp

```diff
@@ -107,6 +107,18 @@ class cluster
         std::chrono::milliseconds update_rate,
         uint64_t                  min_measured) -> bool;
 
+
+    /**
+     * Enables constant speculative executions for requests sent to the Cluster.
+     * @param delay Controls the length of time before sending a speculative request
+     * @param max_executions The maximum number of speculative requests to send.
+                             This should not be higher than your max replication factor.
+     * @return True if the heartbeat intervals were updated.
+     */
+    auto cluster::speculative_execution(
+        std::chrono::milliseconds delay,
+        int                       max_executions) -> bool;
```

Comment:
Can we make this type maybe like uint16_t?  The docs don't say negative numbers are valid, you can just static_cast<int>(max_executions) when passing into the cassandra library

Reply:
Seems reasonable. Think I got that one.